src/headerUtils.js

```diff
@@ -153,29 +154,31 @@ module.exports = function(env) {
       throw new Error('mapData: Invalid batchSize.');
     }
 
-    var ix = env.coroutine.mapDataFetch ?
+    // var ix = env.coroutine.mapDataFetch ?
+    //     env.coroutine.mapDataFetch(data, batchSize, a) :
+    //     null;
+    var _data = env.coroutine.mapDataFetch ?
         env.coroutine.mapDataFetch(data, batchSize, a) :
-        null;
+        data;
```

Comment:
No problem. We can definitely return both from mapDataFetch if needed. I just wonder if what you've said about the observation being called with an address is already implemented in the codebase? Maybe I misunderstood something but when I changed the mapDataFetch I've seen that it's only called from Elbo and that the `ix` are used only inside `mapData`, `cpsMapData` and `elbo.mapDataFetch`. It seems to me that these will continue to work without returning indices from fetch.. Do I miss something or did you mean that we should return indices to support future additions to webppl?